File: navigation/src/main/java/com/example/navigation/stepsEngine/field/IsDeviceConnectedField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.payment.FlowState;

public class IsDeviceConnectedField extends Field {

    @Override
    public Boolean get(FlowState flowState) {
        return flowState.getPos().isConnected();
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/IsDeviceConnectedField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState
import kotlinx.android.parcel.Parcelize

@Parcelize
class IsDeviceConnectedField : Field() {
    override operator fun get(flowState: FlowState): Boolean? {
        return flowState.pos.isConnected
    }

    override fun getId(): String {
        return FieldName.IS_DEVICE_CONNECTED
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}